flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/data/AbstractTestFlinkAvroReaderWriter.java

```diff
@@ -1,182 +0,0 @@
-      for (int i = 0; i < expectedRecords.size(); i++) {
-        assertThat(rows).hasNext();
-        TestHelpers.assertRowData(schema.asStruct(), flinkSchema, expected.next(), rows.next());
-      }
-      assertThat(rows).isExhausted();
-    }
-
-    File rowDataFile = File.createTempFile("junit", null, temp.toFile());
-    assertThat(rowDataFile.delete()).isTrue();
-
-    // Write the expected RowData into AVRO file, then read them into Record and assert with the
-    // expected RowData list.
-    try (FileAppender<RowData> writer =
-        Avro.write(Files.localOutput(rowDataFile))
-            .schema(schema)
-            .createWriterFunc(ignore -> new FlinkAvroWriter(flinkSchema))
-            .build()) {
-      writer.addAll(expectedRows);
-    }
-
-    try (CloseableIterable<Record> reader =
-        Avro.read(Files.localInput(rowDataFile))
-            .project(schema)
-            .createReaderFunc(DataReader::create)
-            .build()) {
-      Iterator<RowData> expected = expectedRows.iterator();
-      Iterator<Record> records = reader.iterator();
-      for (int i = 0; i < expectedRecords.size(); i += 1) {
-        assertThat(records).hasNext();
-        TestHelpers.assertRowData(schema.asStruct(), flinkSchema, records.next(), expected.next());
-      }
-      assertThat(records).isExhausted();
-    }
-  }
-
-  private Record recordNumType(
-      int id,
-      int intV,
-      float floatV,
-      double doubleV,
-      long date,
-      long time,
-      long timestamp,
-      long bigint,
-      double decimal) {
-    Record record = GenericRecord.create(SCHEMA_NUM_TYPE);
-    record.setField("id", id);
-    record.setField("int", intV);
-    record.setField("float", floatV);
-    record.setField("double", doubleV);
-    record.setField(
-        "date", DateTimeUtil.dateFromDays((int) new Date(date).toLocalDate().toEpochDay()));
-    record.setField("time", new Time(time).toLocalTime());
-    record.setField("timestamp", DateTimeUtil.timestampFromMicros(timestamp * 1000));
-    record.setField("bigint", bigint);
-    record.setField("decimal", BigDecimal.valueOf(decimal));
-    return record;
-  }
-
-  @Test
-  public void testNumericTypes() throws IOException {
-    List<Record> expected =
-        ImmutableList.of(
-            recordNumType(
-                2,
-                Integer.MAX_VALUE,
-                Float.MAX_VALUE,
-                Double.MAX_VALUE,
-                Long.MAX_VALUE,
-                1643811742000L,
-                1643811742000L,
-                1643811742000L,
-                10.24d),
-            recordNumType(
-                2,
-                Integer.MIN_VALUE,
-                Float.MIN_VALUE,
-                Double.MIN_VALUE,
-                Long.MIN_VALUE,
-                1643811742000L,
-                1643811742000L,
-                1643811742000L,
-                10.24d));
-
-    writeAndValidate(SCHEMA_NUM_TYPE, expected);
```

Comment:
This might be an interesting edge case to test in the `TestData`?
Min/max values for different columns? Could be especially interesting for timestamps/date etc

Reply:
These cases are tested in the random data generator, so this is duplication.